### PR TITLE
better debugging in GetSocket

### PR DESCRIPTION
**Pull request by maxtaco**
- made this small cleanup when i was adding debugging for fixing CORE-3212

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits ad7ae3791089...634a50e6fb07 introduced some visual changes on linux:
 * Changed: **unlock_folders-Single** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1468003345783-ad7ae3791089-634a50e6fb07-linux/unlock_folders-Single-ad7ae3791089.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1468003345783-ad7ae3791089-634a50e6fb07-linux/unlock_folders-Single-634a50e6fb07.png)  
   unlock_folders-Single comparison

File: go/libkb/socket.go [hunk 1]
```diff
@@ -92,6 +94,7 @@ func (g *GlobalContext) GetSocket(clearError bool) (net.Conn, rpc.Transporter, b
 	if sw.err != nil && clearError {
 		g.SocketWrapper = nil
 	}
+	err = sw.err
```
Comment (patrickxb) [inline go/libkb/socket.go]: you can skip this and just return sw.err...it will still work in the defer.

https://play.golang.org/p/nCeiu3QFLb

**Comment by patrickxb** [comment]
LGTM

File: go/libkb/socket.go (hunk 1 above)
Comment (maxtaco) [inline go/libkb/socket.go]: But what about `Trace`?

Comment (maxtaco) [inline go/libkb/socket.go]: oh wow!

Comment (maxtaco) [inline go/libkb/socket.go]: good to know...

**Pull request merged**